### PR TITLE
Add new checksum refreshing infrastructure

### PR DESCRIPTION
**Pull request by staticfloat**
This changes the `contrib/refresh_bb_tarballs.sh` into a
`contrib/refresh_checksums.mk` which can be run in parallel _much_
faster than the original script thanks to the new `checksum-foo` targets
for all checksummable targets.

This will allow us to build a CI bot that automatically does this for
all PRs, hopefully noticing cases where the checksums are
missing/extraneous ones have not been pruned.

### CODE REVIEW

**Comment by staticfloat** [comment]
The proposed CI bot will do, essentially,

```
rm -rf deps/checksums
make -f contrib/refresh_checksums.mk -j${NPROC}
[ -z $(git status --porcelain) ]
```

The happy path is that the checksums all get recreated instantly.  However, if some checksums fail to be recreated, this means the user should have pruned those checksums, or if there are new checksums, the user should have added those checksums.  In this manner, I hope to avoid Fredrik having to post another Bernie Sanders meme at me.  ;)

File: deps/blas.mk [hunk 1]
```diff
@@ -130,6 +129,7 @@ clean-openblas:
 
 get-openblas: $(OPENBLAS_SRC_FILE)
 extract-openblas: $(BUILDDIR)/$(OPENBLAS_SRC_DIR)/source-extracted
+checksum-openblas: get-openblas
```
Comment (vtjnash) [inline deps/blas.mk:132]:
```suggestion
```

File: contrib/refresh_checksums.mk [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# These are the projects currently using BinaryBuilder; both GCC-expanded and non-GCC-expanded:
+BB_PROJECTS=mbedtls libssh2 nghttp2 mpfr curl libgit2 pcre libuv unwind dsfmt objconv p7zip zlib suitesparse openlibm
+BB_GCC_EXPANDED_PROJECTS=openblas
+BB_CXX_EXPANDED_PROJECTS=gmp llvm
+# These are non-BB source-only deps
+NON_BB_PROJECTS=patchelf mozillacert lapack
+
+# Convert `openblas` to `OPENBLAS`
+define upper
+$(shell echo $(1) | tr '[:lower:]' '[:upper:]')
```
Comment (vtjnash) [inline contrib/refresh_checksums.mk:30]:
```suggestion
$(shell echo $(1) | tr 'a-z' 'A-Z')
```
You aren't actually changing lower to upper in a user locale, but just changing ASCII chars

**Review by vtjnash — APPROVED** [review]
Wouldn't this be easier as a direct bash script, without the Makefile generator, since you haven't actually defined any real targets?  :joy:

**Comment by StefanKarpinski** [comment]
I've been thinking for a while that we should take a lot of the intense functionality that's crammed into our makefiles and turn them into plain old scripts instead and simplify the makefiles. Of course, I also think we should stop supporting in-tree builds of our deps and only support BB or system builds with a separate repo for building our own copies of all the deps from source.

**Comment by staticfloat** [comment]
> Wouldn't this be easier as a direct bash script, without the Makefile generator, since you haven't actually defined any real targets? 😂

Ironically, I started with a direct `bash` script, but the parallelism here is so important that I ended up just using `make` so that I can easily `-j` this.  I was halfway to building my own semaphore/FIFO-based job queuing system based on [this SO](https://unix.stackexchange.com/a/216475) when I stopped myself and just used make.

**Pull request merged**